### PR TITLE
HL-85

### PR DESCRIPTION
**Pull request by xiangzhe0812**
[Link to ticket](https://dltx.atlassian.net/jira/software/projects/HL/boards/82?selectedIssue=HL-85)

unit test for market contract

1. comment out settleMarket.

Screenshot 2022-11-14 at 10 25 17 am

### CODE REVIEW

File: contracts/Market.sol [hunk 1]
```diff
@@ -238,40 +242,42 @@ contract Market is Ownable, IMarket {
...
-
-                if (_bets[index].propositionId == propositionId) {
-                    _settle(index, true);
-                } else {
-                    _settle(index, false);
-                }
-            }
-        }
-    }
+    // function settleMarket(
```
Comment (OptimusOpus) [inline contracts/Market.sol:255]: Are we no longer using this function, or will it be added back in the future?

Comment (xiangzhe0812) [reply contracts/Market.sol:255]: Talked to Barry, not gonna use for now, but not sure for the future, so I commented out.

Comment (bitcoinbrisbane) [reply contracts/Market.sol:255]: We should have settle market

File: contracts/Market.sol [hunk 2]
```diff
@@ -303,7 +312,8 @@ contract Market is Ownable, IMarket {
         _;
     }
 
-    // event Claimed(address indexed worker, uint256 amount);
+    event Claimed(address indexed worker, uint256 amount);
```
Comment (bitcoinbrisbane) [inline contracts/Market.sol:315]: Lets remove this.  No one calls it

File: tests/market.test.ts [hunk 3]
```diff
@@ -99,6 +89,16 @@ describe("Market", () => {
       .connect(carol)
       .approve(market.address, ethers.constants.MaxUint256);
 
+    //Should get 0 odds if vault has ZERO assets
```
Comment (bitcoinbrisbane) [inline tests/market.test.ts:92]:
```suggestion
    // Should get 0 odds if vault has ZERO assets
```

File: tests/market.test.ts [hunk 4]
```diff
@@ -99,6 +89,16 @@ describe("Market", () => {
...
+    const wager = ethers.utils.parseUnits("100", USDT_DECIMALS);
+    const odds = ethers.utils.parseUnits("5", ODDS_DECIMALS);
+    const propositionId = ethers.utils.formatBytes32String("1");
+    expect(await market.getOdds(wager, odds, propositionId)).to.equal(0);
+    //Should get 0 potential payout if vault has Zero odds
```
Comment (bitcoinbrisbane) [inline tests/market.test.ts:97]:
```suggestion
    // Should get 0 potential payout if vault has Zero odds
```

File: tests/market.test.ts [hunk 5]
```diff
@@ -239,6 +244,10 @@ describe("Market", () => {
       ethers.utils.parseUnits("650", USDT_DECIMALS),
       "Vault should have $650 USDT"
     );
+
+    //Should get expiry after back bet
```
Comment (bitcoinbrisbane) [inline tests/market.test.ts:248]:
```suggestion
    // Should get expiry after back bet
```

**Review by bitcoinbrisbane — CHANGES_REQUESTED** [review]
Few minor things

File: contracts/Market.sol (hunk 1 above)
Comment (xiangzhe0812) [reply contracts/Market.sol:255]: Talk to Barry again, we have ticket https://dltx.atlassian.net/jira/software/projects/HL/boards/82/backlog?selectedIssue=HL-153 to refactor the market contract, current settle market logic need to be change, I think we should comment out for now until we fixed the contract then redo unit test.

**Review by bitcoinbrisbane — APPROVED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]

**Review by OptimusOpus — APPROVED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]

**Pull request merged**